BUG: Make scalar.squeeze accept axis arg

toslunar opened the pull request:
Fix #11384.

toslunar committed:
Make scalar.squeeze accept axis arg

shoyer commented:
Please adds tests to verify that this fix works.

eric-wieser commented:
This changes `np.float64(1).squeeze()` to return `array(1.0)` instead of `np.float64(1.0)`. Is this acceptable?

toslunar committed:
Add scalar squeeze tests

toslunar committed:
Fix doc on squeeze in arrays.scalars

toslunar commented:
@shoyer I added a test to `test_regression.py` since I've not found tests on method of scalar types.  Is there a better file to add to?
@eric-wieser I thought I made that bug from your comment, but it seems working correctly.  I also added a test on it.  Besides, I found `x.squeeze() is not x` after this change.

eric-wieser commented on numpy/core/tests/test_regression.py:
```diff
@@ -2375,3 +2375,19 @@ def test_structarray_title(self):
+                continue
+            sctype = np.dtype(ch).type
+            scvalue = sctype(values.get(ch, 3))
+            for axis in [None, ()]:
```
Could add `0` and `-1` here too - surprisingly they're supported

shoyer replied:
I think there's a good case that supporting 0 and -1 on 0d arrays/scalars is a bug.

seberg replied:
True, but most likely the issue should be deprecated separately.

seberg replied:
But, still could add it here really. Just also add a comment that we may want to deprecate it (just so that someone who does it is not scared away by a test).

eric-wieser commented on doc/source/reference/arrays.scalars.rst:
```diff
@@ -274,7 +274,6 @@ The exceptions to the above rules are given below:
    generic
    generic.__array__
    generic.__array_wrap__
-   generic.squeeze
```
I think removing this might cause the documentation for it to disappear. @mattip?

charris replied:
Yes, I think this is needed.

toslunar replied:
Thanks!  I reverted the change.

charris commented:
Test failure was bogus.

toslunar committed:
Revert "Fix doc on squeeze in arrays.scalars"

This reverts commit f4e2480dc6fc3bc000d4a9943f1ccc7ad05d5214.

eric-wieser commented:
I take it back - I don't think there's any point leaving `generic.squeeze` in the generated documentation:

https://docs.scipy.org/doc/numpy/reference/generated/numpy.generic.squeeze.html

charris commented:
The documentation isn't helpful, OTOH, the entry is in a list of exceptions. I suppose the list could be made and ordinary list.

charris commented:
@shoyer Could you open an issue about deprecating 0 and -1?

charris commented:
Thanks @toslunar .
